app/controllers/admin/categories_controller.rb

```diff
@@ -24,13 +24,21 @@ def destroy
   private
 
   def new_or_edit
+    if params[:action] == "new"
+      @category = Category.new
+    elsif params[:action] == "edit" && params[:id]
```

Comment:
Thought: Does the params[:action] in this `elsif` as well as the one below change the conditional, or could you just use params[:id]